src/mux-analytics.brs

```diff
@@ -535,6 +584,122 @@ function muxAnalytics() as Object
+          m._viewTotalUpscaling = m._safeAdd(m._viewTotalUpscaling, upscale * timeDiff)
+          m._viewTotalDownscaling = m._safeAdd(m._viewTotalDownscaling, downscale * timeDiff)
+        end if
+      end if
+      if videoSegment.width <> Invalid AND videoSegment.height <> Invalid AND videoSegment.segBitrateBps <> Invalid
+        if m._lastSourceWidth <> Invalid AND m._lastSourceWidth <> videoSegment.width OR m._lastSourceHeight <> Invalid AND m._lastSourceHeight <> videoSegment.height OR m._lastVideoSegmentBitrate <> Invalid AND m._lastVideoSegmentBitrate <> videoSegment.segBitrateBps
+          m._addEventToQueue(m._createEvent("renditionchange", { video_source_width : videoSegment.width, video_source_height : videoSegment.height, video_source_bitrate : videoSegment.segBitrateBps }))
+        end if
+      end if
+      m._lastSourceWidth = videoSegment.width
+      m._lastSourceHeight = videoSegment.height
+      m._lastVideoSegmentBitrate = videoSegment.segBitrateBps
+      m._lastPlayheadPosition = Int(videoSegment.segStartTime * 1000)
+    end if
+  end sub
+
+  prototype.videoDownloadedSegmentChangeHandler = sub(videoSegment as object)
+    if m._segmentRequestCount = Invalid then m._segmentRequestCount = 0
+    m._segmentRequestCount++
+    if videoSegment <> Invalid
+      props = {}
+      if videoSegment.segType <> Invalid
+        if videoSegment.segType = 0
+          props.request_type = "media"
+        else if videoSegment.segType = 1
+          props.request_type = "audio"
+        else if videoSegment.segType = 2
+          props.request_type = "video"
+        else if videoSegment.segType = 3
+          props.request_type = "captions"
+        end if
+      end if
+      if videoSegment.downloadDuration <> Invalid
+        props.request_media_duration = videoSegment.downloadDuration
+        date = m._getDateTime()
+        now = 0# + date.AsSeconds() * 1000.0#  + date.GetMilliseconds()
+        props.request_response_end = FormatJson(now)
+        resultMilliseconds = now - videoSegment.downloadDuration
+        props.request_start = FormatJson(resultMilliseconds)
+      end if
+      if videoSegment.segUrl <> Invalid
+        props.request_hostname = m._getHostname(videoSegment.segUrl)
+      end if
+      if videoSegment.status <> Invalid
+        if videoSegment.status = 0
+          if videoSegment.segSize <> Invalid
+            props.request_bytes_loaded = videoSegment.segSize
+          end if
+          if videoSegment.width <> Invalid
+            props.request_video_width = videoSegment.width
+          end if
+          if videoSegment.height <> Invalid
+            props.request_video_height = videoSegment.height
+          end if
+          if m.video <> Invalid AND m.video.content <> Invalid
+            if m.video.contentIsPlaylist
+              props.video_source_is_live = m.video.content.getChild(m.video.contentIndex).live
```

Comment:
interesting, I'm not sure that we need to check this every time a segment is downloaded, because this shouldn't change, right? I'd probably just remove this unless there's a specific reason

Reply:
Yes, it won't change. will create a new variable which holds the value of live status. If you don't need it in `requestcomplete` event, happy to remove it.